app/controllers/tasks_controller.rb

```diff
@@ -31,12 +35,40 @@ def create
     end
   end
 
+  def edit
+    @task 
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to tasks_path, notice: 'Task was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def toggle_status
+    @task.update(status: @task.status == 1 ? 0 : 1)
```

Comment:
check aasm lib in here https://github.com/aasm/aasm to apply it

Reply:
done